src/App.tsx

```diff
@@ -14,11 +14,129 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+type Props = {};
+
+type State = {
+  goods: string[];
+  isVisibleStartButton: boolean;
+  isVisibleOtherButtons: boolean;
+  isReversed: boolean;
+  sortBy: string;
+};
+
+export class App extends React.Component<Props, State> {
+  state: State = {
+    goods: [],
+    isVisibleStartButton: true,
+    isVisibleOtherButtons: false,
+    isReversed: false,
+    sortBy: '',
+  };
+
+  showGoods = () => {
+    this.setState({
+      goods: goodsFromServer,
+      isVisibleStartButton: false,
+      isVisibleOtherButtons: true,
+    });
+  };
+
+  reverseGoods = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortGoodsByName = () => {
+    this.setState({ sortBy: 'Name' });
+  };
+
+  resetGoods = () => {
+    this.setState(() => ({
+      isReversed: false,
+      sortBy: '',
+    }));
```

Comment:
If you don't use prevState in functional `setState()`, you can just:
```suggestion
    this.setState({ isReversed: false, sortBy: ''});
```